controllers/helmrepository_controller.go

```diff
@@ -123,7 +123,16 @@ func (r *HelmRepositoryReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *HelmRepositoryReconciler) SetupWithManagerAndOptions(mgr ctrl.Manager, opts HelmRepositoryReconcilerOptions) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&sourcev1.HelmRepository{}).
-		WithEventFilter(predicate.Or(predicate.GenerationChangedPredicate{}, predicates.ReconcileRequestedPredicate{})).
+		WithEventFilter(
+			predicate.And(
+				predicate.Or(
+					predicate.NewPredicateFuncs(HelmRepositoryTypeFilter(sourcev1.HelmRepositoryTypeDefault)),
+					// an empty type field defaults to handling the repo as traditional HTTP Helm repo
+					predicate.NewPredicateFuncs(HelmRepositoryTypeFilter("")),
```

Comment:
Wouldn't it be better to make the filter adhere to the predicate interface instead of wrapping it? The type can be configured in the struct.

Reply:
I'm not sure I understand what the benefit of a dedicated type implementing the Predicate interface would be.

Reply:
It is easier for me to later find something that's structured as a predicate, than to think of a function that exists somewhere.

In addition, adhering to an interface often kind of forces you into writing tests. Of which at present, I do not see any.

Reply:
I asked @pjbgf for another opinion on this so we can settle it. However, I do not think this change is blocking. Could be taken care of as a refinement as part of the dependency support.

Reply:
For improved clarity, I think that being able to pass on the filter is better than calling in a func, which is then passed on to another func, which then returns something which implements the predicate interface.

The same goes for `predicate.NewPredicateFuncs(HelmRepositoryTypeFilter(""))`. If it was a predicate in itself, with a clear name, it would probably not need a comment to explain it.